### PR TITLE
dyff: init at 1.4.3

### PR DESCRIPTION
**Pull request by edlimerkaj**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/dyff/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ lib, buildGoModule, fetchFromGitHub}:
+
+let
+  version = "1.4.3";
+in
+
+buildGoModule rec {
+  inherit version;
```
Comment (marius851000) [inline pkgs/development/tools/dyff/default.nix:8]: You can directly use `version = "1.4.3"` here, and remove the let ... in as `rec` mean (in `buildGoModule rec`) mean that child can directly access property of this attrset (so `rev = "v${version}"` will work)

**Review by marius851000 — CHANGES_REQUESTED** [review]
Tested on x86-64 (nixos). It work, and can convert a random json to yaml with `dyff yaml <json path>`

File: pkgs/development/tools/dyff/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      changes from one version to another by only showing the parts of a YAML
+      file that change.
+
+      Each difference is referenced by its location in the YAML document by
+      using either the Spruce or go-patch path syntax.
+    '';
+    homepage = "https://github.com/homeport/dyff";
+    license = licenses.mit;
+    maintainers = with maintainers; [ edlimerkaj ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dyff/default.nix:36]:
```suggestion
```
default for buildGoModule

File: pkgs/development/tools/dyff/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{ lib, buildGoModule, fetchFromGitHub}:
+
+let
+  version = "1.4.3";
+in
+
+buildGoModule rec {
+  inherit version;
+
+  pname = "dyff";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dyff/default.nix:10]:
```suggestion
buildGoModule rec {
  pname = "dyff";
  version = "1.4.3";
```

File: pkgs/development/tools/dyff/default.nix (hunk 1 above)
Comment (edlimerkaj) [reply pkgs/development/tools/dyff/default.nix]: Thank you for your feedback. I have made the suggested change.

File: pkgs/development/tools/dyff/default.nix (hunk 3 above)
Comment (edlimerkaj) [reply pkgs/development/tools/dyff/default.nix]: Thank you for your feedback. I have made the suggested changes.

**Comment by 06kellyjac** [comment]
Please set the PR title to `dyff: init at 1.4.3`

File: pkgs/development/tools/dyff/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{ lib, buildGoModule, fetchFromGitHub}:
+
+buildGoModule rec {
+
+  pname = "dyff";
+  version = "1.4.3";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dyff/default.nix:7]:
```suggestion
buildGoModule rec {
  pname = "dyff";
  version = "1.4.3";

  src = fetchFromGitHub {
```

File: pkgs/development/tools/dyff/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      file that change.
+
+      Each difference is referenced by its location in the YAML document by
+      using either the Spruce or go-patch path syntax.
+    '';
+    homepage = "https://github.com/homeport/dyff";
+    license = licenses.mit;
+    maintainers = with maintainers; [ edlimerkaj ];
+
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dyff/default.nix:32]:
```suggestion
    maintainers = with maintainers; [ edlimerkaj ];
  };
```

File: pkgs/development/tools/dyff/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "homeport";
+    repo = "dyff";
+    rev = "v${version}";
+    sha256 = "0r1nfwglyw8b46n17bpmgscfmjhjsbk83lgkpm63ysy0h5r84dq8";
+  };
+
+  vendorSha256 = "12mirnw229x5jkzda0c45vnjnv7fjvzf0rm3fcy5f3wza6hkx6q7";
+
+  subPackages = [ "cmd/dyff" "pkg/dyff" "internal/cmd" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dyff/default.nix:16]:
```suggestion
  subPackages = [
    "cmd/dyff"
    "pkg/dyff"
    "internal/cmd"
  ];
```

**Pull request merged**